### PR TITLE
Major health analyzer readability improvements

### PR DESCRIPTION
**Pull request by googobbug**
## About The Pull Request

Makes quite a few changes to make health analyzers much easier to read:
- Useless "minor X damage" text removed
- Non-verbose mode still displays overall numbers instead of removing the entire damage value table (who uses this?)
- Heart attack indicator is now bold and put inside the actual scan text
- Missing organs displayed in organ section
- Large amounts of redundant information removed from the advanced scanner seriously it was so bad
- Recommended treatment text removed from health scan wounds readout. Use the dedicated wound mode if you want to see it.
- Normal scanner can now detect missing eyes and ears, as it could already detect organ damage in them. Blindness/deafness detection are still advanced-scanner exclusive.

And the big one: chemical allergies are now displayed when performing a reagent scan!

## Why It's Good For The Game

Makes the health analyzer slightly less impenetrable, makes the advanced scanner way less of an eyesore, puts important information in visible spots so people can actually see missing organs, heart attacks, and allergies. 

Seriously, just look at it!

image
image
image

## Changelog

:cl: Bumtickley00
qol: Health analyzers have had significant readability improvements!
qol: Health analyzers display allergy information when performing a reagent scan.
/:cl:

### CODE REVIEW

**Comment by newfren** [comment]
the missing organ section is great

**Comment by GoldenAlpharex** [comment]
That's really cool.

**Comment by FernandoJ8** [comment]
GOD, THANK YOU.
It's hard to overstate how big of a pain it is to be unable to tell if an organ is missing or not. This is awesome.

**Comment by Farquaar** [comment]
Yikes, what did you do to my boy Cesar

**Comment by AdmiralPancakes1** [comment]
I assure you he deserved it

**Comment by exymian** [comment]
How do we scan for body temp&blood level ? Can you detail out what kind of information you removed that you feel is useless ?

Don't get me wrong this should be done and is a good idea

**Comment by newfren** [comment]
body temp/blood level is on the images shown

**Comment by AdmiralPancakes1** [comment]
the only info I removed was stuff that was already displayed elsewhere in the same scan

**Comment by exymian** [comment]
thats my bad i thought the images were before/after type so i was confused

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -264,58 +271,20 @@ GENE SCANNER
...
 							<td><font color='#33ccff'><b>[CEILING(oxy_loss,1)]</b></font></td></tr>"
 
-			for(var/o in damaged)
-				var/obj/item/bodypart/org = o //head, left arm, right arm, etc.
-				dmgreport += "<tr><td><font color='#cc3333'>[capitalize(org.name)]:</font></td>\
-								<td><font color='#cc3333'>[(org.brute_dam > 0) ? "[CEILING(org.brute_dam,1)]" : "0"]</font></td>\
-								<td><font color='#ff9933'>[(org.burn_dam > 0) ? "[CEILING(org.burn_dam,1)]" : "0"]</font></td></tr>"
+			if(mode == SCANNER_VERBOSE)
+				for(var/o in damaged)
+					var/obj/item/bodypart/org = o //head, left arm, right arm, etc.
```
Comment (tralezab) [inline code/game/objects/items/devices/scanners.dm:276]: **NOTE:** this suggestion changes the var name, you'll need to change other uses of `org` > `limb`. org is not a good var name!

```suggestion
				for(var/obj/item/bodypart/limb as anything in damaged)
```

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -165,17 +165,17 @@ GENE SCANNER
 		mob_status = span_alert("<b>Deceased</b>")
 		oxy_loss = max(rand(1, 40), oxy_loss, (300 - (tox_loss + fire_loss + brute_loss))) // Random oxygen loss
 
+	render_list += "[span_info("Analyzing results for [M]:")]\n<span class='info ml-1'>Overall status: [mob_status]</span>\n"
+	
+	SEND_SIGNAL(M, COMSIG_LIVING_HEALTHSCAN, render_list, advanced, user, mode)
```
Comment (tralezab) [inline code/game/objects/items/devices/scanners.dm:170]: while you're here, go to the top of this proc and rename the var `M`, it's not a descriptive var

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -184,36 +184,18 @@ GENE SCANNER
...
-			render_list += "<span class='alert ml-1'>Subject lacks a heart.</span>\n"
-		if (!(TRAIT_NOBREATH in the_dudes_species.species_traits) && !the_dude.getorganslot(ORGAN_SLOT_LUNGS))
-			render_list += "<span class='alert ml-1'>Subject lacks lungs.</span>\n"
-		if (!(TRAIT_NOMETABOLISM in the_dudes_species.species_traits) && !the_dude.getorganslot(ORGAN_SLOT_LIVER))
-			render_list += "<span class='alert ml-1'>Subject lacks a liver.</span>\n"
-		if (!(NOSTOMACH in the_dudes_species.species_traits) && !the_dude.getorganslot(ORGAN_SLOT_STOMACH))
-			render_list += "<span class='alert ml-1'>Subject lacks a stomach.</span>\n"
 
 	if(iscarbon(M))
 		var/mob/living/carbon/C = M
```
Comment (tralezab) [inline code/game/objects/items/devices/scanners.dm:201]: make this var more descriptive, please

File: code/game/objects/items/devices/scanners.dm [hunk 4]
```diff
@@ -237,17 +219,42 @@ GENE SCANNER
...
 
 	if (HAS_TRAIT(M, TRAIT_IRRADIATED))
 		render_list += "<span class='alert ml-1'>Subject is irradiated. Supply toxin healing.</span>\n"
 
 	if(advanced && M.hallucinating())
 		render_list += "<span class='info ml-1'>Subject is hallucinating.</span>\n"
+		
+	//Eyes and ears
+	if(advanced && iscarbon(M))
+		var/mob/living/carbon/C = M
```
Comment (tralezab) [inline code/game/objects/items/devices/scanners.dm:231]: make this var more descriptive, please

File: code/game/objects/items/devices/scanners.dm [hunk 5]
```diff
@@ -237,17 +219,42 @@ GENE SCANNER
...
+		if(istype(eyes))
+			if(C.is_blind())
+				render_list += "<span class='alert ml-2'>Subject is blind.\n</span>"
+			else if(HAS_TRAIT(C, TRAIT_NEARSIGHT))
+				render_list += "<span class='alert ml-2'>Subject is nearsighted.\n</span>"
 
 	// Body part damage report
-	if(iscarbon(M) && mode == SCANNER_VERBOSE)
+	if(iscarbon(M))
 		var/mob/living/carbon/C = M
```
Comment (tralezab) [inline code/game/objects/items/devices/scanners.dm:258]: make this var more descriptive, please

File: code/game/objects/items/devices/scanners.dm [hunk 6]
```diff
@@ -264,58 +271,20 @@ GENE SCANNER
...
-			if(eyes.damage > 30)
-				message += "\n<span class='alert ml-2'>Subject has severe eye damage.</span>"
-			else if(eyes.damage > 20)
-				message += "\n<span class='alert ml-2'>Subject has significant eye damage.</span>"
-			else if(eyes.damage)
-				message += "\n<span class='alert ml-2'>Subject has minor eye damage.</span>"
-		render_list += "<span class='info ml-1'>Eye status:</span>[message == "" ? "\n<span class='info ml-2'>Healthy.</span>" : message]\n"
-
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
```
Comment (tralezab) [inline code/game/objects/items/devices/scanners.dm:284]: make this var more descriptive, please

File: code/game/objects/items/devices/scanners.dm [hunk 7]
```diff
@@ -458,9 +451,16 @@ GENE SCANNER
 
 		if(M.has_status_effect(/datum/status_effect/eigenstasium))
 			render_list += "<span class='notice ml-1'>Subject is temporally unstable. Stabilising agent is recommended to reduce disturbances.</span>\n"
-
+		
+		for(var/datum/quirk/quirky in M.quirks)
```
Comment (tralezab) [inline code/game/objects/items/devices/scanners.dm:455]:
```suggestion
		for(var/datum/quirk/quirky as anything in M.quirks)
```

**Review by tralezab — CHANGES_REQUESTED** [review]
some var cleanup while you're here, and use of `as anything`

**Comment by AdmiralPancakes1** [comment]
Alright, should be good to go.
Additional changes aside from style fixes:
Made the wounds readout on the health even more compact
image
Fixed some weird indent stuff that possibly made cybernetics not show up when scanning plasmamen (?)
"No addictions" text removed from chem scan, as the lack of big red text should be enough of an indicator

File: code/game/objects/items/devices/scanners.dm [hunk 8]
```diff
@@ -264,147 +271,136 @@ GENE SCANNER
...
+		var/mob/living/carbon/carbontarget = target
+		var/list/wounded_parts = carbontarget.get_wounded_bodyparts()
 		for(var/i in wounded_parts)
 			var/obj/item/bodypart/wounded_part = i
-			render_list += "<span class='alert ml-1'><b>Warning: Physical trauma[LAZYLEN(wounded_part.wounds) > 1? "s" : ""] detected in [wounded_part.name]</b>"
+			render_list += "<span class='alert ml-1'><b>Physical trauma[LAZYLEN(wounded_part.wounds) > 1? "s" : ""] detected in [wounded_part.name]</b>"
 			for(var/k in wounded_part.wounds)
 				var/datum/wound/W = k
-				render_list += "<div class='ml-2'>Type: [W.name]\nSeverity: [W.severity_text()]\nRecommended Treatment: [W.treat_text]</div>\n" // less lines than in woundscan() so we don't overload people trying to get basic med info
+				render_list += "<div class='ml-2'>[W.name] ([W.severity_text()])</div>" // less lines than in woundscan() so we don't overload people trying to get basic med info
```
Comment (Mothblocks) [inline code/game/objects/items/devices/scanners.dm:367]: Bring back the `treat_text`, wound scans honestly don't need to exist and I'd like to remove them someday. The venn diagram of people who know about wound scan mode and don't already know how to solve wounds is very minimal, and switching between them is clunky.

File: code/game/objects/items/devices/scanners.dm [hunk 9]
```diff
@@ -264,147 +271,136 @@ GENE SCANNER
...
-	if(iscarbon(M))
-		var/mob/living/carbon/C = M
-		var/list/wounded_parts = C.get_wounded_bodyparts()
+	if(iscarbon(target))
+		var/mob/living/carbon/carbontarget = target
+		var/list/wounded_parts = carbontarget.get_wounded_bodyparts()
 		for(var/i in wounded_parts)
 			var/obj/item/bodypart/wounded_part = i
-			render_list += "<span class='alert ml-1'><b>Warning: Physical trauma[LAZYLEN(wounded_part.wounds) > 1? "s" : ""] detected in [wounded_part.name]</b>"
+			render_list += "<span class='alert ml-1'><b>Physical trauma[LAZYLEN(wounded_part.wounds) > 1? "s" : ""] detected in [wounded_part.name]</b>"
```
Comment (Mothblocks) [inline code/game/objects/items/devices/scanners.dm:364]:
```suggestion
			render_list += "<span class='alert ml-1'><b>Physical trauma[LAZYLEN(wounded_part.wounds) > 1 ? "s" : ""] detected in [wounded_part.name]</b>"
```

File: code/game/objects/items/devices/scanners.dm [hunk 10]
```diff
@@ -264,147 +271,136 @@ GENE SCANNER
...
 
-			for(var/o in damaged)
-				var/obj/item/bodypart/org = o //head, left arm, right arm, etc.
-				dmgreport += "<tr><td><font color='#cc3333'>[capitalize(org.name)]:</font></td>\
-								<td><font color='#cc3333'>[(org.brute_dam > 0) ? "[CEILING(org.brute_dam,1)]" : "0"]</font></td>\
-								<td><font color='#ff9933'>[(org.burn_dam > 0) ? "[CEILING(org.burn_dam,1)]" : "0"]</font></td></tr>"
+			if(mode == SCANNER_VERBOSE)
+				for(var/obj/item/bodypart/limb as anything in damaged)
+					dmgreport += "<tr><td><font color='#cc3333'>[capitalize(limb.name)]:</font></td>\
+									<td><font color='#cc3333'>[(limb.brute_dam > 0) ? "[CEILING(limb.brute_dam,1)]" : "0"]</font></td>\
```
Comment (Mothblocks) [inline code/game/objects/items/devices/scanners.dm:277]: Don't use tab alignment, this is banned in the style guide.
Split this into three `dmgreport += ""` instead.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
> Non-verbose mode still displays overall numbers instead of removing the entire damage value table (who uses this?)

No one, and like all verbs it should be axed.

**Comment by AdmiralPancakes1** [comment]
wow yeah idk what I did to this branch but all my maps are conflicting and now it can't find the python executable

will fix after finals probably

**Comment by AdmiralPancakes1** [comment]
oh god damn it

**Comment by AdmiralPancakes1** [comment]
reopening on new branch

**Pull request closed without merging**